### PR TITLE
Update lading to Rust 1.74.0

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit updates our rust toolchain to 1.74 and switches its profile from 'default' to 'complete', meaning rust-analyzer et al are installed now by default for every lading developer.

### Related issues

REF SMPTNG-94

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.71.0"
-profile = "default"
+channel = "1.74.0"
+profile = "complete"
```
Comment (GeorgeHahn) [inline rust-toolchain.toml:3]: Does this give a benefit? I know the vscode rust analyzer extension handles this automatically.

Comment (blt) [reply rust-toolchain.toml:3]: It's nice for emacs. I figured it'd be a no-op in the worst case.

**Comment by blt** [comment]
> Please update the rust version in dockerfiles as well

Oops, completely forgot. Good looking out.

**Review by GeorgeHahn — APPROVED** [review]
✔️ once dockerfiles are updated

**Pull request merged**